document/core/exec/instructions.rst

```diff
@@ -1087,94 +1072,74 @@ Table Instructions
 
 7. Let :math:`\X{ea}` be the :ref:`element address <syntax-elemaddr>` :math:`F.\AMODULE.\MIELEMS[x]`.
 
-8. Assert: due to :ref:`validation <valid-table.init>`, :math:`S.\SELEM[\X{ea}]` exists.
+8. Assert: due to :ref:`validation <valid-table.init>`, :math:`S.\SELEMS[\X{ea}]` exists.
 
-9. Let :math:`\X{elem}^?` be the optional :ref:`element instance <syntax-eleminst>` :math:`S.\SELEM[\X{ea}]`.
+9. Let :math:`\X{elem}` be the :ref:`element instance <syntax-eleminst>` :math:`S.\SELEMS[\X{ea}]`.
 
 10. Assert: due to :ref:`validation <valid-table.init>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-11. Pop the value :math:`\I32.\CONST~cnt` from the stack.
+11. Pop the value :math:`\I32.\CONST~n` from the stack.
 
 12. Assert: due to :ref:`validation <valid-table.init>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-13. Pop the value :math:`\I32.\CONST~src` from the stack.
+13. Pop the value :math:`\I32.\CONST~s` from the stack.
 
 14. Assert: due to :ref:`validation <valid-table.init>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-15. Pop the value :math:`\I32.\CONST~dst` from the stack.
-
-16. If :math:`cnt = 0`, then:
+15. Pop the value :math:`\I32.\CONST~d` from the stack.
 
-    a. Return.
-
-17. If :math:`\X{elem}^? = \epsilon`, then:
+16. If :math:`s + n` is larger than the length of :math:`\X{elem}.\EIELEM` or :math:`d + n` is larger than the length of :math:`\X{tab}.\TIELEM`, then:
 
     a. Trap.
 
-18. If :math:`cnt = 1`, then:
-
-    a. Push the value :math:`\I32.\CONST~dst` to the stack.
+17. If :math:`n = 0`, then:
 
-    b. If `src` is larger than the length of :math:`\X{elem}.\EIINIT`, then:
-
-       i. Trap.
-
-    c. Let :math:`\funcelem` be the :ref:`function element <syntax-funcelem>` :math:`\X{elem}.\EIINIT[src]`.
-
-    d. Push the value :math:`\funcelem` to the stack.
+    a. Return.
 
-    e. Execute the instruction :math:`\TABLESET`.
+18. Let :math:`\funcelem` be the :ref:`function element <syntax-funcelem>` :math:`\X{elem}.\EIELEM[s]`.
 
-    f. Return.
+19. Push the value :math:`\I32.\CONST~d` to the stack.
 
-19. Push the value :math:`\I32.\CONST~dst` to the stack.
+20. Push the value :math:`\funcelem` to the stack.
```

Comment:
It may be more clear to reorder to 18, 20, then 19 so that the use of the funcelem value is right next to the definition.

Additionally, it's a bit odd to use funcelem here as a value, but I understand that's due to reftypes being a separate proposal. Is the intention for this to be an administrative instruction like table.get/set in the mean time?

Reply:
Yes, exactly. This is all a temporary hack. We screwed up the dependencies for these proposals.